packages/fxa-settings/src/models/Account.ts

```diff
@@ -1239,36 +1242,31 @@ export class Account implements AccountData {
     return totp;
   }
 
-  async verifyTotpSetupCode(code: string) {
+  async verifyTotpSetupCodeWithJwt(code: string) {
     await this.withLoadingStatus(
-      this.authClient.verifyTotpSetupCode(sessionToken()!, code, {})
+      this.authClient.verifyTotpSetupCodeWithJwt(
+        this.getCachedJwtByScope('2fa'),
+        code,
+        {}
+      )
     );
   }
 
-  async completeTotpSetup(service?: string) {
-    try {
-      await this.withLoadingStatus(
-        this.authClient.completeTotpSetup(
-          sessionToken()!,
-          service ? { service } : {}
-        )
-      );
-      // Only update local cache if the server-side setup completes successfully
-      await this.refresh('recoveryPhone');
-      const cache = this.apolloClient.cache;
-      cache.modify({
-        id: cache.identify({ __typename: 'Account' }),
-        fields: {
-          totp(currentTotp) {
-            return { ...currentTotp, verified: true };
-          },
+  async completeTotpSetupWithJwt() {
+    await this.withLoadingStatus(
+      this.authClient.completeTotpSetupWithJwt(this.getCachedJwtByScope('2fa'))
+    );
+    const cache = this.apolloClient.cache;
```

Comment:
Just double checking, the `await this.refresh(...)` call was dropped. Is that intentional?